web/lib/HabitatSidebar.js

```diff
@@ -151,9 +167,32 @@ class HabitatSidebar extends HTMLElement {
     window.addEventListener('hashchange', onNavigate, false);
     onNavigate();
 
+    this.wrapActions();
     this.update();
   }
 
+  wrapActions() {
+    wrapListener(this.querySelector('a#topup'), async () => {
+      this.updateDOM('Top Up Gas Tank');
+    });
+    wrapListener(this.querySelector('a#deposit'), async () => {
+      this.updateDOM('Deposit');
+    });
+    wrapListener(this.querySelector('a#withdraw'), async () => {
+      this.updateDOM('Withdraw');
+    });
+  }
+
+  // Update navigation action
+  updateDOM(action) {
```

Comment:
You can also make it simpler by passing arguments to the element via `#habitat-account,foo,bar,baz`.
Act on those changes in `attributeChangedCallback`.

At the moment that would break button highlighting but will push a fix asap.

Reply:
Was looking to achieve something similar when i started. Hopefully changes reflect what you envision